perf_bench_test.go

```diff
@@ -24,127 +24,241 @@ package main
 
 import (
 	"context"
+	"encoding/csv"
 	"flag"
+	"fmt"
+	"os"
 	"strconv"
+	"strings"
 	"sync"
+	"sync/atomic"
 	"testing"
 	"time"
 
+	"github.com/ease-lab/vhive/profile"
 	log "github.com/sirupsen/logrus"
 	"github.com/stretchr/testify/require"
 )
 
 var (
-	isColdStart     = flag.Bool("coldStart", false, "Profile cold starts (default is false)")
 	vmNum           = flag.Int("vm", 2, "The number of VMs")
-	targetReqPerSec = flag.Int("requestPerSec", 2, "The target number of requests per second")
-	executionTime   = flag.Int("executionTime", 2, "The execution time of the benchmark in seconds")
+	targetReqPerSec = flag.Int("requestPerSec", 10, "The target number of requests per second")
+	executionTime   = flag.Int("executionTime", 1, "The execution time of the benchmark in seconds")
+	funcNames       = flag.String("funcNames", "helloworld", "Name of the functions to benchmark")
+
+	perfExecTime = flag.Float64("perfExecTime", 0, "The execution time of perf command in seconds (run `perf stat` if bigger than 0)")
+	perfInterval = flag.Uint64("perfInterval", 100, "Print count deltas every N milliseconds")
+	perfEvents   = flag.String("perfEvents", "", "Perf events")
+)
+
+const (
+	averageExecutionTime  = "average-execution-time"
+	realRequestsPerSecond = "real-requests-per-second"
 )
 
+func TestBenchMultiVMRequestPerSecond(t *testing.T) {
+	baseRPS := getCPUIntenseRPS()[*funcName]
+	*funcNames = *funcName
+
+	for i := 4; i <= 20; i += 4 {
+		*vmNum = i
+		*targetReqPerSec = *vmNum * baseRPS
+		*funcNames = *funcName
+		testName := fmt.Sprintf("%d VMs,%d RPS", *vmNum, *targetReqPerSec)
+		t.Run(testName, TestBenchRequestPerSecond)
+	}
+
+	profile.CSVPlotter(getOutFile("benchRPS.csv"), *benchDir)
```

Comment:
the filename must be parameterized. in fact, it's better to parameterize the folder with output files while keep the filenames the same.

Reply:
I do not get how to change it.

Reply:
just make the file or folder name the runtime parameter

Reply:
Done